plugins/modules/lambda_event.py

```diff
@@ -252,13 +267,32 @@ def lambda_event_stream(module, client):
     else:
         module.fail_json(msg="Source parameter 'source_arn' is required for stream event notification.")
 
-    # check if optional sub-parameters are valid, if present
-    batch_size = source_params.get("batch_size")
-    if batch_size:
-        try:
-            source_params["batch_size"] = int(batch_size)
-        except ValueError:
-            module.fail_json(msg=f"Source parameter 'batch_size' must be an integer, found: {batch_size}")
+    if state == "present":
+        if module.params["event_source"].lower() == "sqs":
+            # Default 10. For standard queues the max is 10,000. For FIFO queues the max is 10.
+            if not source_params.get("batch_size"):
+                source_params["batch_size"] = 10
+            else:
+                if source_params["source_arn"].endswith(".fifo"):
+                    if source_params["batch_size"] > 10:
+                        module.fail_json(msg="For FIFO queues the maximum batch_size is 10.")
+                    if source_params.get("maximum_batching_window_in_seconds"):
+                        module.fail_json(
+                            msg="maximum_batching_window_in_seconds is not supported by Amazon SQS FIFO event sources."
+                        )
+                else:
+                    if not (100 <= source_params["batch_size"] <= 10000):
+                        module.fail_json(msg="For standard queue batch_size must be between 100 and 10000.")
```

Comment:
I'm not sure I like the depth of these nested if-statements this feels like the kind of tangled nest of logic that can come back to bite us.

Reply:
I tried to implement the checks for batch_size and maximum_batching_window_in_seconds as explained in the documentation https://boto3.amazonaws.com/v1/documentation/api/1.26.78/reference/services/lambda.html#Lambda.Client.create_event_source_mapping. However, I also cleaned a bit now.

What the documentation was reporting as default value for batch_size was wrong. It shouldn't always default to 100. I fixed that and I also added the breaking_change section in the changelog. backport-7 label is also gone.

Reply:
> I tried to implement the checks for batch_size and maximum_batching_window_in_seconds as explained in the documentation

If it needs to be complex, then it needs to be complex.  I was thinking more that it probably needs to be pulled out into its own function that we can unit test: I don't think we're likely to create integration tests that go through the various combinations that are possible here...

Reply:
I added unit tests instead. Thanks.